instance_dire_maul.cpp

```diff
@@ -0,0 +1,315 @@
+                    {
+                        Creature* mob = instance->GetCreature(_crystalCreatureGUIDs[i][j]);
+                        if (mob && mob->IsAlive())
+                        {
+                            _mobAlive = true;
+                            break;
+                        }
+                    }
+                }
+
+                if (!_mobAlive && go->GetGoState() == GO_STATE_READY) // if all stored creatures are dead and go state is ready
+                {
+                    HandleGameObject(ObjectGuid::Empty, false, go);
+
+                    switch (go->GetEntry())
+                    {
+                        case GO_CRYSTAL_01:
+                            SetBossState(DATA_CRYSTAL_01, DONE);
+                            go->SetGoState(GO_STATE_ACTIVE);
+                            break;
+                        case GO_CRYSTAL_02:
+                            SetBossState(DATA_CRYSTAL_02, DONE);
+                            go->SetGoState(GO_STATE_ACTIVE);
+                            break;
+                        case GO_CRYSTAL_03:
+                            SetBossState(DATA_CRYSTAL_03, DONE);
+                            go->SetGoState(GO_STATE_ACTIVE);
+                            break;
+                        case GO_CRYSTAL_04:
+                            SetBossState(DATA_CRYSTAL_04, DONE);
+                            go->SetGoState(GO_STATE_ACTIVE);
+                            break;
+                        case GO_CRYSTAL_05:
+                            SetBossState(DATA_CRYSTAL_05, DONE);
+                            go->SetGoState(GO_STATE_ACTIVE);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            }
+
+            if (GetBossState(DATA_CRYSTAL_01) == DONE && GetBossState(DATA_CRYSTAL_02) == DONE && GetBossState(DATA_CRYSTAL_03) == DONE &&
+                GetBossState(DATA_CRYSTAL_04) == DONE && GetBossState(DATA_CRYSTAL_05) == DONE)
+            {
+                // if all crystals are done, we set encounter forcefield to done
+                SetBossState(DATA_FORCEFIELD, DONE);
+                // activate forcefield to make it disappear
+                if (GameObject* ffield = instance->GetGameObject(_forcefieldGUID))
+                    ffield->SetGoState(GO_STATE_ACTIVE);
+                // remove previously set non attackable flag
+                if (Creature* immo = instance->GetCreature(_immoGUID))
+                    immo->RemoveUnitFlag(UNIT_FLAG_NON_ATTACKABLE);
+            }
+        }
+
+        bool IsImmoDead()
+        {
+            Creature* immo = instance->GetCreature(_immoGUID);
+            if (immo != nullptr && !immo->IsAlive())
+            {
+                Creature* tortheldrin = instance->GetCreature(_tortheldrinGUID);
+                if (tortheldrin != nullptr)
+                    tortheldrin->SetFaction(FACTION_ENEMY);

```

Comment:
```suggestion
                if (Creature* tortheldrin = instance->GetCreature(_tortheldrinGUID))
                    tortheldrin->SetFaction(FACTION_ENEMY);

```

Reply:
Done as of f0530402887f2e91769d47adb45736bca45809e7.